packages/compass-workspaces/src/components/index.tsx

```diff
@@ -47,25 +73,24 @@ const WorkspacesWithSidebar: React.FunctionComponent<
   activeTab,
   activeTabCollectionInfo,
   onActiveWorkspaceTabChange,
-  initialConnectionInfo,
+  sidebar,
+  modals,
 }) => {
   const onChange = useRef(onActiveWorkspaceTabChange);
   onChange.current = onActiveWorkspaceTabChange;
   useEffect(() => {
     onChange.current(activeTab, activeTabCollectionInfo);
   }, [activeTab, activeTabCollectionInfo]);
   return (
-    <div className={horizontalSplitStyles}>
-      <div className={sidebarStyles}>
-        <CompassSidebarPlugin
```

Comment:
To avoid having a recursive dependency (sidebar depends on workspaces service -> workspaces depend on sidebar for rendering it), changed the direct dependency to a slot component interface, similar to what we now do for modals